owasp-top10-2017-apps/a6/stegonography/README.md

```diff
@@ -0,0 +1,113 @@
+
+The main goal of this project is to discuss how **Security Misconfiguration** vulnerabilities can be exploited and to encourage developers to send secDevLabs Pull Requests  on how they would mitigate these flaws.
+
+## Setup
+
+To start this intentionally **insecure application**, you will need [Docker][Docker Install] and [Docker Compose][Docker Compose Install]. After forking [secDevLabs](https://github.com/globocom/secDevLabs), you must type the following commands to start:
+
+```sh
+cd secDevLabs/owasp-top10-2017-apps/a6/stegonography
+```
+
+```sh
+make install
+```
+
+Then simply visit [localhost:10006][App] ! 😆
+
+## Get to know the app 🦕
+
+To properly understand how this application works, you can follow these simple steps:
+
+- Try adding an hidden message to an image.
+- Try hiding a message with a password! 
+- Decrypt a friends message!
+
+## Attack narrative
+
+Now that you know the purpose of this app, what could possibly go wrong? The following section describes how an attacker could identify and eventually find sensitive information about the app or it's users. We encourage you to follow these steps and try to reproduce them on your own to better understand the attack vector! 😜
+
+### Vulnerability identification  👀
+
+#### Error stack traces are output to end users
+
+An attacker, when trying to enumerate available pages on the application, could come across a verbose error stack trace with potentially sensitive information that could compromise the app. An example of a verbose error stack trace is as shown by the image below:
+
+<p align="center">
+    <img src="images/stack_trace.png"/>
+</p>
+
+#### Default username and passwords are being used in a production environment 
+
+Using [Dirb] to enumerate existing pages on the application and hide the "Not Found" with the `-N 401` flag, it's possible to find what seems to be a login page, as pointed in the image below:
+
+```sh
+dirb http://localhost:10006 -N 401
+```
+
+<p align="center">
+    <img src="images/dirb_result.png"/>
+</p>
+
+By visiting `http://localhost:10006/login` we get to the following screen:
+
+<p align="center">
+    <img src="images/login_page.png"/>
+</p>
+
+A quick guess with the credentials `admin:admin` and we can successfully log-in to the application and get to the admin's control-panel, as shown by the image below:
+
+<p align="center">
+    <img src="images/admin_page.png"/>
+</p>
+
+#### Verbose session token gives away unnecessary information
+
+After logging in to the application, it's possible to see it sets a session token: `jwtSessionToken`. As shown by the following image:
+
+<p align="center">
+    <img src="images/jwt_token.png"/>
+</p>
+
+Even though using verbose tokens might not pose a straight forward security vulnerability, on certain apps it might give away the hint an attacker needs to compromise it's security. 🕵️‍
+
```

Comment:
I miss the `Attack/Exploitation phase` here as this is an attack narrative 😃

Reply:
In this particular app this issue does not pose a security vulnerability. I thought that maybe an exploration phase might not be necessary.

Reply:
Just added it!